Document HTML Components and Clarify Page Types (#1538)
withastro/docs: src/pages/en/core-concepts/astro-components.md

Astro detects these JavaScript client-side imports and then builds, optimizes, and adds the JS to the page automatically.

```astro

> 

Reviewer: Nit: this’ll get stripped out anyway, but no need for the extra newline.
```suggestion
```

// ESM import
<script>
  import './some-external-script.js';
</script>
```

> ## HTML Components
> 
> Astro supports using `.html` files instead of `.astro` files for components. These can be imported like Astro components, but don't support Astro features. As such:
> - They don't support frontmatter, server-side imports, or dynamic expressions.
> - Any `<script>` tags are left unbundled, treated as if they had `is:inline`. 
> - They can only [reference assets that are in the `public/` folder.](/en/guides/images/#public)
> - [Slots](/en/core-concepts/astro-components/#slots) work as they do in Astro components. If you'd like to use [`Web Components slots feature`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot) instead, add `is:inline` to your `<slot>` element.
> 
> You may want to use HTML Components if you're migrating from a site built without a framework, or if you want to ensure that your component has no dynamic features.

Reviewer 2: ```suggestion
## HTML Components

Astro supports importing and using `.html` files as components or placing these files within the `src/pages` subdirectory as pages. You may want to use HTML components if you're reusing code from an existing site built without a framework, or if you want to ensure that your component has no dynamic features.

HTML components must contain only valid HTML, and therefore lack key Astro component features:
- They don't support frontmatter, server-side imports, or dynamic expressions.
- Any `<script>` tags are left unbundled, treated as if they had `is:inline`. 
- They can only [reference assets that are in the `public/` folder.](/en/guides/images/#public)

:::note
A [`<slot />` element](/en/core-concepts/astro-components/#slots) inside an HTML component will work as it does in Astro component. In order to use the [HTML Web Component Slot](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot) element instead, add `is:inline` to your `<slot>` element.
:::
```

Not sure all of the above is entirely the correct nuance, but here's my thinking behind some changes above that I'm not married to, but are illustrative of what came to mind:

- took out "instead" of astro files because presumably you can use both? (If not, then maybe that should be a bigger deal?)
- spent a LOT of time thinking about this "migrating an older site" and am leaning towards the specific "and therefore wanting to reuse some existing code" angle of it. Migrating an older site doesn't necessarily mean keeping the same code, and therefore doesn't necessarily mean needing an HTML component (so, want to avoid the confusion of someone thinking they need to/should use them?). I think these HTML components become useful *if* you are bringing that old code over *and wanting as little of a rewrite as possible*, just not simply by virtue of the fact you're migrating an older, html-based site?
- put the reason you might want to use them up front, so people know why they're even reading this or might need this
- assuming this is true, I'd preface the "things you can't do that you can in Astro" with the brief reminder that these are just straight HTML, therefore...
- OMG SLOTS LOL. I *think* the distinction to make here is that there's an "Astro Slot" and a regular HTML slot element, and the cautionary note needs to be if you're using a slot, it's gonna be treated like an Astro slot by default. BUT, if you want it to be treated like the regular web slot element, that's cool: add `is:inline`. I also thought about making this a note since it's maybe worth a call out that esp if you're bringing old code over that has a slot, you should make sure you know it's NOT going to be that default web component? Not sure, but wanted to illustrate the option.

UPDATED after reading the Pages edit to include mention of these being pages, too. I did not cram in "must contain a full proper document of HTML" but am not opposed if you want to find a way to do that.

Author: I agree with every point. The slot disclaimer does make sense as a note rather than part of the limitations list!

</script>
```

## HTML Components

Astro supports importing and using `.html` files as components or placing these files within the `src/pages` subdirectory as pages. You may want to use HTML components if you're reusing code from an existing site built without a framework, or if you want to ensure that your component has no dynamic features.

HTML components must contain only valid HTML, and therefore lack key Astro component features:
- They don't support frontmatter, server-side imports, or dynamic expressions.
- Any `<script>` tags are left unbundled, treated as if they had `is:inline`. 

> - They can only [reference assets that are in the `public/` folder.](/en/guides/images/#public)

Reviewer 2: ```suggestion
- They can only [reference assets that are in the `public/` folder](/en/guides/images/#public).
```
